src/ReachSets/discretize.jl

```diff
@@ -1,97 +1,185 @@
+The initial value problem returned by this function consists of the set `Ω0`
+together with the coefficients matrix ``ϕ = e^{Aδ}`` and a transformed
+set of inputs if `U` is non-empty.
+
+In the literature, the method to obtain `Ω0` is called the *approximation model*
+and different alternatives have been proposed. See the argument `approx_model`
+for available options. For the reference to the original papers, see the docstring
+of each method.
+
+The transformation described above allows to do dense time reachability, i.e.
+such that the trajectories of the given continuous system are included in the computed
+flowpipe of the discretized system. In particular, if you don't want to consider
+bloating, i.e. discrete-time reachability, use `approx_model="nobloating"`.
+
+Several methods to compute the matrix exponential are availabe. Use `expm_method`
+to select one. For very large systems (~10000×10000), computing the full matrix
+exponential is very expensive hence it is preferable to compute the action
+of the matrix exponential over vectors when needed. Use the option
+`expm_method="lazy"` for this.
 """
-function discretize(cont_sys::InitialValueProblem{<:AbstractContinuousSystem},
+function discretize(𝑆::InitialValueProblem{<:AbstractContinuousSystem},
                     δ::Float64;
                     approx_model::String="forward",
-                    pade_expm::Bool=false,
-                    lazy_expm::Bool=false,
-                    lazy_sih::Bool=true)::InitialValueProblem{<:AbstractDiscreteSystem}
+                    expm_method::String="base",
+                    sih_method::String="lazy")
+
+    if !isaffine(S)
+        throw(ArgumentError("`discretize` is only implemented for affine ODEs"))
+    end
 
     if approx_model in ["forward", "backward"]
-        return discr_bloat_interpolation(cont_sys, δ, approx_model, pade_expm,
-                                         lazy_expm, lazy_sih)
+        return discr_bloat_interpolation(𝑆, δ, approx_model, expm_method, sih_method)
     elseif approx_model == "firstorder"
-        return discr_bloat_firstorder(cont_sys, δ)
+        return _discretize_first_order(𝑆, δ, expm_method)
     elseif approx_model == "nobloating"
-        return discr_no_bloat(cont_sys, δ, pade_expm, lazy_expm)
+        return discr_no_bloat(𝑆, δ, expm_method)
     else
-        error("The approximation model is invalid")
+        throw(ArgumentError("the approximation model $approx_model is unknown"))
+    end
+end
+
+function exp_Aδ(A::AbstractMatrix, δ::Float64, expm_method="base")
+    if expm_method == "base"
+        return expmat(Matrix(A*δ))
+    elseif expm_method == "lazy"
+        return SparseMatrixExp(A*δ)
+    elseif expm_method == "pade"
+        return padm(A*δ)
+    else
+       throw(ArgumentError("the exponentiation method $expm_method is unknown"))
     end
 end
 
 """
-    bloat_firstorder(cont_sys, δ)
+    _discretize_first_order(𝑆, δ, [p], [expm_method])
```

Comment:
I would not use a leading underscore if you refer to this function in the docs.

Reply:
i've removed the leading underscore and added `discretize_....` algorithms to the docs.